TripPlanner/View Controllers/TripViewController.m

```diff
@@ -0,0 +1,259 @@
+    }];
+}
+
+- (IBAction)onSave:(id)sender {
+    NSMutableArray *images = [[NSMutableArray alloc] init];
+    [images addObject:[self getPFFileFromImage:self.tripImageView.image]];
+    [Trip postUserTrip:self.guestUsernames withImages:images withDescription:self.descriptionTextView.text withTitle:self.titleField.text withLocation:self.locationField.text withStartDate:self.startDatePicker.date withEndDate:self.endDatePicker.date withGuests:self.guests withController:self];
+}
+
+- (IBAction)onCancel:(id)sender {
+    // reset fields
+    self.titleField.text = nil;
+    self.locationField.text = nil;
+    self.guestsField.text = nil;
+    self.guestList.text = @"Guests: ";
+    self.guests = [[NSMutableArray alloc] init];
+    self.startDatePicker.date = [NSDate date];
+    self.endDatePicker.date = [NSDate date];
+    self.descriptionTextView.text = nil;
+    self.tripImageView.image = [UIImage imageNamed:@"image_placeholder"];
+}
+
+- (UIAlertController *) createSourceTypeAlert {
+    // create a camera choice action
+    UIAlertAction *cameraAction = [UIAlertAction actionWithTitle:@"Camera"
+                                                           style:UIAlertActionStyleDefault
+                                                         handler:^(UIAlertAction * _Nonnull action) {
+        [self createImagePicker:@"Camera"];
+    }];
+    
+    // create a photo album choice action
+    UIAlertAction *albumAction = [UIAlertAction actionWithTitle:@"Album"
+                                                          style:UIAlertActionStyleDefault
+                                                        handler:^(UIAlertAction * _Nonnull action) {
+        [self createImagePicker:@"Album"];
+    }];
+    
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle: @"Photo Upload"
+                                                                   message: @"Choose upload source"
+                                                            preferredStyle:(UIAlertControllerStyleAlert)];
+    
+    // add the camera action to the alertController
+    [alert addAction:cameraAction];
+    
+    // add the album action to the alert controller
+    [alert addAction:albumAction];
+    
+    return alert;
+    
+}
+
+- (PFFileObject *)getPFFileFromImage: (UIImage * _Nullable)image {
+    
+    // check if image is not nil
+    if (!image) {
+        return nil;
+    }
+    
+    NSData *imageData = UIImagePNGRepresentation(image);
+    // get image data and check if that is not nil
+    if (!imageData) {
+        return nil;
+    }
+    
+    return [PFFileObject fileObjectWithName:@"image.png" data:imageData];
+}
+
+- (BOOL)hasGuest: (NSString *)username guests: (NSMutableArray *)guests {
+    for(NSString *guestUsername in guests) {
+        if ([username isEqualToString:guestUsername]) {
+            return YES;
+        }
+    }
+    return NO;
+}
+
+- (IBAction)onAdd:(id)sender {
```

Comment:
would be great to group the "onAdd" like functions and others together, to make reading the code easier. You can even consider splitting some of the functionality into another class and use that class here instead of having all the functions on the same controller.